src/net.c

```diff
@@ -673,6 +673,9 @@ static int net__bind_interface(struct mosquitto__listener *listener, struct addr
 						memcpy(&((struct sockaddr_in6 *)rp->ai_addr)->sin6_addr,
 								&((struct sockaddr_in6 *)ifa->ifa_addr)->sin6_addr,
 								sizeof(struct in6_addr));
+
+						((struct sockaddr_in6 *)rp->ai_addr)->sin6_scope_id = ((struct sockaddr_in6 *)ifa->ifa_addr)->sin6_scope_id;
```

Comment:
Can you explain why the memcpy call in line 673 isn't sufficient to copy this value already?

Reply:
Is your question why only copying the in6_addr is not enough? That's because for link-local addresses they are only valid for a specific interface, you can specify which interface the address belongs to with the scope id.

If your question is why I didn't group everything into a single memcpy call. That was for readability.